metadata/src/main/java/org/apache/kafka/controller/ConfigurationControlManager.java

```diff
@@ -66,6 +69,7 @@ public class ConfigurationControlManager {
     private final TimelineHashMap<ConfigResource, TimelineHashMap<String, String>> configData;
     private final Map<String, Object> staticConfig;
     private final ConfigResource currentController;
+    private final MinIsrConfigUpdatePartitionHandler minIsrConfigUpdatePartitionHandler;
```

Comment:
maybe more of a question for someone with more code ownership of the quorum controller code, but I wonder if it would be preferable to handle generating the replication control manager records in the `QuorumController.incrementalAlterConfigs`. That would also make it a bit easier to handle `validateOnly` which we are not currently handling.